starter-code/ShoppingListWithDetailView/app/src/main/java/ly/generalassemb/drewmahrt/shoppinglistdetailview/DetailActivity.java

```diff
@@ -0,0 +1,40 @@
+package ly.generalassemb.drewmahrt.shoppinglistdetailview;
+
+import android.os.Bundle;
+import android.support.v7.app.AppCompatActivity;
+import android.widget.TextView;
+
+public class DetailActivity extends AppCompatActivity {
+    public static final String ITEM_ID_KEY = "itemIDKey";
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_detail);
+
+        int id = getIntent().getIntExtra(ITEM_ID_KEY, -1);
+
+        if (id == -1) {
+            finish();
```

Comment:
Instead of just kicking the user back to the previous activity if there is an error, the better solution would be to show some sort of error first.